### ISSUE TITLE
Can not get the root directory contents

### ISSUE THREAD

**Issue by linw1995** [perry-mitchell/webdav-client#185]
Hi,
I am using the demo from the README.md file. But it not working

```js
const { createClient } = require("webdav");

const client = createClient(
    "https://dav.jianguoyun.com/dav",
    {
        username: "username@example.com",
        password: "password",
    }
);
client
    .getDirectoryContents("/") // error 404
    // .getDirectoryContents("") // success
    .then(contents => console.log(contents))
    .catch(error => console.error(error));
```

https://github.com/perry-mitchell/webdav-client/blob/493636bb0cea307b4bc1d6ab2520ee7df9db00be/source/interface/directoryContents.js#L12

this line of code may cause this problem, it will tail an excessive slash when using "/".

```js
    status: 404,
    statusText: 'Not Found',
    headers: {
      server: 'nginx',
      date: 'Wed, 11 Dec 2019 13:38:16 GMT',
      'content-type': 'text/xml; charset=UTF-8',
      'content-length': '231',
      connection: 'close',
      pragma: 'no-cache',
      'cache-control': 'no-cache'
    },
    config: {
      url: 'https://dav.jianguoyun.com/dav//',
```

**Comment by perry-mitchell** [comment]
Thanks for the detailed issue. The `url-join` library shouldn't add extra slashes:

```javascript
const joinURL = require("url-join");
console.log(joinURL("https://example.com/", "/some-page/", "/"));
// https://example.com/some-page/
```

But after running another test I can see that there are indeed multiple slashes being added. I'll investigate more.

**Comment by perry-mitchell** [comment]
Ok, the issue is happening when multiple paths are attached using single slashes: `joinURL("https://example.com", "/", "/")`.

**Comment by perry-mitchell** [comment]
Made a PR - will merge and release today :) Thanks!

**Comment by linw1995** [comment]
Thanks! @perry-mitchell

**Comment by perry-mitchell** [comment]
Released in 2.10.1 :)
